heat/core/linalg/basics.py

```diff
@@ -752,24 +770,277 @@ def matmul(a: DNDarray, b: DNDarray, allow_resplit: bool = False) -> DNDarray:
 DNDarray.__matmul__ = lambda self, other: matmul(self, other)
 
 
-def norm(a: DNDarray) -> float:
+def matrix_norm(
+    x: DNDarray,
+    axis: Optional[Tuple[int, int]] = None,
+    keepdims: bool = False,
+    ord: Optional[Union[int, Literal["fro", "nuc"]]] = None,
+) -> DNDarray:
     """
-    Return the vector norm (Frobenius norm) of vector ``a``.
+    Computes the matrix norm of an array.
 
     Parameters
     ----------
-    a : DNDarray
+    x : DNDarray
+        Input array
+    axis : tuple, optional
+        Both axes of the matrix. If `None` 'x' must be a matrix. Default: `None`
+    keepdims : bool, optional
+        Retains the reduced dimension when `True`. Default: `False`
+    ord : int, 'fro', 'nuc', optional
+        The matrix norm order to compute. If `None` the Frobenius norm (`'fro'`) is used. Default: `None`
+
+    See Also
+    --------
+    norm
+        Computes the vector or matrix norm of an array.
+    vector_norm
+        Computes the vector norm of an array.
+
+    Notes
+    -----
+    The following norms are supported:
+
+    =====  ============================
+    ord    norm for matrices
+    =====  ============================
+    None   Frobenius norm
+    'fro'  Frobenius norm
+    'nuc'  nuclear norm
+    inf    max(sum(abs(x), axis=1))
+    -inf   min(sum(abs(x), axis=1))
+    1      max(sum(abs(x), axis=0))
+    -1     min(sum(abs(x), axis=0))
+    =====  ============================
```

Comment:
did you check this in the compiled docs? i think that this might make things messy

Reply:
It works as expected locally

![image](https://user-images.githubusercontent.com/18899420/129694519-e083a78b-85d6-4d96-9340-934fe6e530ff.png)